packages/core/admin/admin/src/content-manager/pages/ListView/components/BulkActionButtons/tests/index.test.js

```diff
@@ -155,3 +187,117 @@ describe('BulkActionsBar', () => {
+  ...render(
+    <ConfirmDialogPublishAll
+      isOpen
+      onConfirm={jest.fn()}
+      onToggleDialog={jest.fn()}
+      isConfirmButtonLoading
+    />,
+    {
+      wrapper({ children }) {
+        const client = new QueryClient({
+          defaultOptions: {
+            queries: {
+              retry: false,
+            },
+          },
+        });
+
+        return (
+          <QueryClientProvider client={client}>
+            <ThemeProvider theme={lightTheme}>
+              <IntlProvider locale="en" messages={{}} defaultLocale="en">
+                <Provider store={store}>
+                  <MemoryRouter>
+                    <Table.Root>{children}</Table.Root>
+                  </MemoryRouter>
+                </Provider>
+              </IntlProvider>
+            </ThemeProvider>
+          </QueryClientProvider>
+        );
+      },
+    }
+  ),
+});
+
+describe('ConfirmDialogPublishAll', () => {
+  beforeAll(() => {
+    server.listen();
+  });
+
+  afterAll(() => {
+    server.close();
+  });
+  it('should show a default message if there are not draft relations', async () => {
+    const { getByText, queryByText } = setupConfirmPublish();
+    await waitFor(() => expect(getByText('Confirmation')).toBeInTheDocument());
+    expect(
+      queryByText('not published yet and might lead to unexpected behavior')
+    ).not.toBeInTheDocument();
+    await waitFor(() =>
+      expect(getByText('Are you sure you want to publish these entries?')).toBeInTheDocument()
+    );
+  });
+  it('should show the warning message with just 1 draft relation and two entries', async () => {
+    server.use(
+      rest.get('*/countManyEntriesDraftRelations', (req, res, ctx) => {
+        return res.once(
+          ctx.status(200),
+          ctx.json({
+            data: 1,
+          })
+        );
+      })
+    );
+
+    const { getByText, getByTestId } = setupConfirmPublish();
+    await waitFor(() => expect(getByText('Confirmation')).toBeInTheDocument());
+    const draftWarning = await getByTestId('draft-rel-warning');
+
+    await waitFor(() =>
+      expect(draftWarning.textContent).toBe('1  relation  out of 2  entries   is ')
+    );
+  });
+  it('should show the warning message with 2 draft relations and two entries', async () => {
+    server.use(
+      rest.get('*/countManyEntriesDraftRelations', (req, res, ctx) => {
+        return res.once(
+          ctx.status(200),
+          ctx.json({
+            data: 2,
+          })
+        );
+      })
+    );
+
+    const { getByText, getByTestId } = setupConfirmPublish();
+    await waitFor(() => expect(getByText('Confirmation')).toBeInTheDocument());
+    const draftWarning = await getByTestId('draft-rel-warning');
+
+    await waitFor(() =>
+      expect(draftWarning.textContent).toBe('2  relations  out of 2  entries   are ')
+    );
+  });
+  it('should not show the Confirmation component if there is an error coming from the API', async () => {
```

Comment:
I think we would benefit from checking the notification shows in the UI :)

Reply:
ok let me try